src/main/java/de/hpi/bpt/argos/api/eventType/EventTypeEndpointImpl.java

```diff
@@ -0,0 +1,435 @@
+                EventQuery[] attributeArray = new EventQuery[attributes.size()];
+                feedback = PersistenceAdapterImpl.getInstance().deleteArtifacts(attributes.toArray(attributeArray));
+                if (!feedback) {
+                    halt(HttpStatusCodes.ERROR, "could not delete corresponding attributes of event type");
+                }
+
+                // delete events
+                List<Event> events = PersistenceAdapterImpl.getInstance().getEventsOfEventType(eventTypeId);
+                EventQuery[] eventArray = new EventQuery[events.size()];
+                feedback = PersistenceAdapterImpl.getInstance().deleteArtifacts(events.toArray(eventArray));
+                if (!feedback) {
+                    halt(HttpStatusCodes.ERROR, "could not delete corresponding events of event type");
+                }
+
+                // delete mappings
+                List<EventEntityMapping> mappings = PersistenceAdapterImpl.getInstance().getEventEntityMappingsForEventType(eventTypeId);
+                EventQuery[] mappingArray = new EventQuery[mappings.size()];
+                feedback = PersistenceAdapterImpl.getInstance().deleteArtifacts(mappings.toArray(mappingArray));
+                if (!feedback) {
+                    halt(HttpStatusCodes.ERROR, "could not delete corresponding mappings of event type");
+                }
+
+                // delete event type
+                feedback = PersistenceAdapterImpl.getInstance().deleteArtifact(eventType,
+                        EventTypeEndpoint.getDeleteEventTypeUri(eventType.getId()));
+                if (!feedback) {
+                    halt(HttpStatusCodes.ERROR, "could not delete event type");
+                }
+
+                response.status(HttpStatusCodes.SUCCESS);
+            } else {
+                response.status(HttpStatusCodes.ERROR);
+                return blockingEventTypes;
+            }
+        }
+        return "";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getEventTypeAttributes(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+        JsonArray jsonTypeAttributes = new JsonArray();
+
+        try {
+            long eventTypeId = getEventTypeId(request);
+            List<TypeAttribute> attributes = PersistenceAdapterImpl.getInstance().getTypeAttributes(eventTypeId);
+
+            for (TypeAttribute attribute : attributes) {
+                JsonObject jsonTypeAttribute = new JsonObject();
+
+                jsonTypeAttribute.addProperty("Id", attribute.getId());
+                jsonTypeAttribute.addProperty("Name", attribute.getName());
+
+                jsonTypeAttributes.add(jsonTypeAttribute);
+            }
+        } catch (Exception e) {
+            logger.error("cannot parse request body to type attributes '" + request.body() + "'");
+            logTrace(e);
+            halt(HttpStatusCodes.ERROR, e.getMessage());

```

Comment:
Use `BAD_REQUEST` instead of `ERROR`, since this is not an internal server error, but an user error